File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionIn.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionIn extends TransformerFunction {

    public TransformerFunctionIn() {
        super(FunctionDescription.of(
            Map.of(
            "value", ArgumentType.of(ArgType.Any).position(0)
            )
        ));
    }

    @Override
    public Object apply(FunctionContext context) {
        var value = context.getJsonElement("value");
        var collection = context.getJsonElementStreamer(null);
        var adapter = context.getAdapter();
        return collection != null && collection.stream().anyMatch(other -> adapter.areEqual(value, other));
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionIn.ts
import TransformerFunction from "./common/TransformerFunction";
import { ArgType } from "./common/ArgType";
import FunctionContext from "./common/FunctionContext";
import { isEqual } from "../JsonHelpers";

class TransformerFunctionIn extends TransformerFunction {
  constructor() {
    super({
      argsSet: [{ name: "value", type: ArgType.Any }],
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const value = await context.getJsonElement("value");
    const collection = await context.getJsonElementStreamer(null);
    return collection !== null && collection.stream().any(other => isEqual(value, other));
  }
}

export default TransformerFunctionIn;
